src/types/web-compat.ts

```diff
@@ -0,0 +1,46 @@
+// eslint-disable
+/**
+ * @module Webcompat Messages Schema
+ * @description
+ * These types are auto-generated from schema files.
+ * scripts/build-types.mjs is responsible for type generation.
+ * See the privacy-configuration repo for the schema files:
+ * https://github.com/duckduckgo/privacy-configuration
+ * **DO NOT** edit this file directly as your changes will be lost.
+ */
+
+/**
+ * Requests, Notifications and Subscriptions from the Web Compat feature
+ */
+export interface WebCompatMessages {
+  notifications: unknown;
+  requests: WebShare;
+  subscriptions: unknown;
+}
+/**
+ * todo: add description for `webShare` message
+ */
+export interface WebShare {
+  method: "webShare";
+  params: WebShareParams;
+  /**
+   * todo: add return type here
+   */
+  result: {
+    [k: string]: unknown;
+  };
+}
+export interface WebShareParams {
+  /**
+   * todo: add description for 'title' field
+   */
+  title?: string;
+  /**
+   * todo: add description for 'url' field
+   */
+  url?: string;
+  /**
+   * todo: add description for 'text' field
+   */
+  text?: string;
+}
```

Comment:
This is an example where in the JSDoc we gave an object a `title` -> That forces the the code generator to make a separate interface and allows you to import it into JS files